File: zoho_post_invoice.py
import requests
import json
from datetime import datetime, timedelta
from zoho_token_manager import get_headers, get_api_domain

organization_id = "892673756"
api_domain = get_api_domain()
headers = get_headers(organization_id)

# Endpoint for creating invoices
url = f"{api_domain}/books/v3/invoices"

# Sample invoice data
today = datetime.now().strftime("%Y-%m-%d")
due_date = (datetime.now() + timedelta(days=30)).strftime("%Y-%m-%d")

invoice_data = {
    "customer_id": "6630269000001113041",  # Customer Zero2Cento from fetched invoices
    "date": today,
    "due_date": due_date,
    "currency_code": "AED",
    "line_items": [
        {
            "name": "Sample Product",
            "description": "This is a test invoice created via API",
            "quantity": 2,
            "rate": 150.0
        },
        {
            "name": "Sample Service",
            "description": "Consulting service",
            "quantity": 1,
            "rate": 300.0
        }
    ],
    "notes": "Thank you for your business!"
}

try:
    response = requests.post(url, headers=headers, json=invoice_data)
    response.raise_for_status()
    result = response.json()
    print("Invoice created successfully!")
    print(json.dumps(result, indent=2))
except requests.exceptions.RequestException as e:
    print(f"Error creating invoice: {e}")
    if response is not None:
        print(f"Response status: {response.status_code}")
        print(f"Response body: {response.text}")

File: zoho_post_invoice.js
const axios = require('axios');
const { getHeaders, getApiDomain } = require('./zoho_token_manager');

async function createInvoice() {
    const organizationId = "892673756";
    const apiDomain = getApiDomain();
    const headers = await getHeaders(organizationId);

    // Endpoint for creating invoices
    const url = `${apiDomain}/books/v3/invoices`;

    // Sample invoice data
    const today = new Date().toISOString().split('T')[0];
    const dueDate = new Date();
    dueDate.setDate(dueDate.getDate() + 30);
    const dueDateStr = dueDate.toISOString().split('T')[0];

    const invoiceData = {
        "customer_id": "6630269000001113041",  // Customer Zero2Cento from fetched invoices
        "date": today,
        "due_date": dueDateStr,
        "currency_code": "AED",
        "line_items": [
            {
                "name": "Sample Product",
                "description": "This is a test invoice created via API",
                "quantity": 2,
                "rate": 150.0
            },
            {
                "name": "Sample Service",
                "description": "Consulting service",
                "quantity": 1,
                "rate": 300.0
            }
        ],
        "notes": "Thank you for your business!"
    };

    try {
        const response = await axios.post(url, invoiceData, { headers });
        const result = response.data;
        console.log("Invoice created successfully!");
        console.log(JSON.stringify(result, null, 2));
        return result;
    } catch (error) {
        console.log(`Error creating invoice: ${error.message}`);
        if (error.response) {
            console.log(`Response status: ${error.response.status}`);
            console.log(`Response body: ${error.response.data}`);
        }
        throw error;
    }
}

// Execute the function if this file is run directly
if (require.main === module) {
    createInvoice();
}

module.exports = { createInvoice };